infrastructures/openshift/src/main/java/org/eclipse/che/workspace/infrastructure/openshift/bootstrapper/OpenshiftBootstrapper.java

```diff
@@ -0,0 +1,115 @@
+import org.eclipse.che.inject.ConfigurationException;
+import org.eclipse.che.workspace.infrastructure.openshift.OpenshiftMachine;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.inject.Inject;
+import javax.inject.Named;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.List;
+
+/**
+ * Bootstraps installers in openshift machine.
+ *
+ * @author Sergii Leshchenko
+ */
+public class OpenshiftBootstrapper extends AbstractBootstrapper {
+    private static final Logger LOG = LoggerFactory.getLogger(OpenshiftBootstrapper.class);
+
+    private static final Gson GSON = new GsonBuilder().disableHtmlEscaping()
+                                                      .create();
+
+    private static final String BOOTSTRAPPER_BASE_DIR = "/tmp/";
+    private static final String BOOTSTRAPPER_DIR      = BOOTSTRAPPER_BASE_DIR + "bootstrapper/";
+    private static final String BOOTSTRAPPER_FILE     = "bootstrapper";
+    private static final String CONFIG_FILE           = "config.json";
+
+    private final String              machineName;
+    private final RuntimeIdentity     runtimeIdentity;
+    private final List<InstallerImpl> installers;
+    private final int                 serverCheckPeriodSeconds;
+    private final int                 installerTimeoutSeconds;
+    private final OpenshiftMachine    openshiftMachine;
+    private final String              downloadBootstrapperLink;
+
+    @Inject
+    public OpenshiftBootstrapper(@Assisted String machineName,
+                                 @Assisted RuntimeIdentity runtimeIdentity,
+                                 @Assisted List<InstallerImpl> installers,
+                                 @Assisted OpenshiftMachine openshiftMachine,
+                                 @Named("che.infra.openshift.che_server_websocket_endpoint_base") String websocketBaseEndpoint,
+                                 @Named("che.infra.openshift.che_server_endpoint") String workspaceMasterEndpoint,
+                                 @Named("che.infra.openshift.bootstrapper.timeout_min") int bootstrappingTimeoutMinutes,
+                                 @Named("che.infra.openshift.bootstrapper.installer_timeout_sec") int installerTimeoutSeconds,
+                                 @Named("che.infra.openshift.bootstrapper.server_check_period_sec") int serverCheckPeriodSeconds,
+                                 EventService eventService) {
+        super(machineName, runtimeIdentity, bootstrappingTimeoutMinutes, websocketBaseEndpoint, eventService);
+        try {
+            URL wsmasterEndpoint = new URL(workspaceMasterEndpoint);
+            this.downloadBootstrapperLink = wsmasterEndpoint.getProtocol() + "://" + wsmasterEndpoint.getHost() + ":" +
+                                            wsmasterEndpoint.getPort() +
+                                            "/agent-binaries/linux_amd64/bootstrapper/bootstrapper";
```

Comment:
Added dedicated property `che.infra.openshift.bootstrapper_binary_url`.

Maybe later we will inject bootstrapper binary in other way but now I have no idea how we can implement it without execution command for downloading like curl or wget.